### ISSUE TITLE
Error with Drivers

### ISSUE THREAD

**Issue by rmiller-ch** [FriendsOfMDT/PSD#194]
I am new to PSD and either I am missing something in the documentation but I have questions about driver support.  My example is I downloaded the drivers from Dells website for a Precision 3460 SFF (cab file). I have created a folder structure in MDT under Out-of-Box drivers as follows Dell \ Precision 3460 \ Windows 11 and imported the drivers into said folder.  I then ran the New-PSDDriverPackage.ps1 script and it created a WIM file name Dell-Precision_3460-Windows_11.wim.  When I run the PSD client task sequence do I need to create a selection profile and modify my task sequence to use it or does that matter anymore.  As it stands now, its not finding the WIM file with the drivers?  If that were to fail, what are the fallback options and how do they work?

**Comment by rmiller-ch** [comment]
Forgot to add that is finds 0 driver packages.

**Comment by MDA400** [comment]
Running into this issue as well. Put the drivers into a folder in Out-Of-Box Drivers that is named exactly what the model shows in the PSDWizard ( **20NX0036US** for example). Trying next with just the driver files straight into the folder that is named the model of the device im imaging (no subfolders).

**Comment by rmiller-ch** [comment]
I created a "Windows 11" folder under Out-of-Box Drivers in MDT with a subfolder of "Precision 3460" and imported all the Dell Precision 3460 drivers.  I am now running the New-PSDDriverPackage.ps1 script and will see what happens.  More to come. Thanks for the tip.

**Comment by MDA400** [comment]
Have a look at this post about modelalias use with configmgr - https://www.deploymentresearch.com/using-modelalias-for-configmgr-driver-management/

Its likely how its being used. There are lines in the ZTIGather.xml file (scripts folder) that explain where it is searching for these files:

<property id="DriverPath" type="string" overwrite="false" description="Folder under PSDResources\DriverPackages where Drivers exists" />

<property id="GenericDriverPath" type="string" overwrite="false" description="Folder under PSDResources\DriverPackages where Generic Drivers exists" />

<property id="FallBackDriverPath" type="string" overwrite="false" description="Folder under PSDResources\DriverPackages where Fallback Drivers exists" />

**Comment by rmiller-ch** [comment]
I also noticed in the Task Sequences I created using the PSD Standard Client Task Sequence, you can set those items manually I guess as they appear as options.  There is an Inject Drivers task but I left it alone and the PSDWizard found the driver wim labeled "Windows_11-Precision_3460.wim" as I created it in the MDT Out-of-Box Drivers area.  It does not seem to follow the naming convention that I thought I read about total driver control, but I could be misunderstanding something.  I am formatting now, hopefully it works, but the machine I am create has a raid controller (Intel RST) and it always fails for me.

**Comment by rmiller-ch** [comment]
Well it seems to find the drivers, but now all computers I try imaging just sit at the Getting Ready screen after the PSDWizard runs once.  Not sure why that is happening.  Ugh

**Comment by MDA400** [comment]
Yeah, documentation is a little out of date (references names of old scripts for example). Got it working by changing **Windows 11 \ %ModelAlias%** to just **Windows 11 \ %Model%** for the "Set DriverPath" step in the task sequence.

Wish i could get past the WinPE step myself, having an issue where the "Make The OS Volume Bootable" step is throwing an application error (0xe0434352) after my custom 23H2 image is transferred to the device. The image applies successfully under normal MDT environment.

As for your "getting ready" stall issue after the WinPE step is done, could be an included driver issue or issues with connectivity to a domain (if you are trying to use domain join in the process). Have had those with normal MDT environment, this link has locations of log files that can be used to identify why it may be stalling.

https://support.microsoft.com/en-us/topic/log-files-that-are-created-when-you-upgrade-to-a-new-version-of-windows-9ec8aa31-0cc1-a0b2-2d98-e9c6714349b9

**Comment by rmiller-ch** [comment]
Damn....the moment you said it I knew that's the problem.  I had this happen before when I was using regular MDT and the account did not have access to add computers to a certain location, same as today.  Fixing that and trying again.

**Comment by PScherling** [comment]
Hey, I had the same issue on my first run. I changed the Values in "Set-DriverPath" from **%ModelAlias%** to **%Model%** and that solved my issue. Just to confirm your solution :)

**Comment by PowerShellCrack** [comment]
This may be a good update to our docs. I did update the docs in the CurrentDev branch; it is in Peer review but hopefully it gets moved to the main branch. Here is the updated doc: https://github.com/FriendsOfMDT/PSD/blob/CurrentDev/Documentation/PowerShell%20Deployment%20-%20Installation%20Guide.md
![image](https://github.com/user-attachments/assets/1dbe728b-e686-432c-a652-b5742d3f70d5)

I created a Navigation for it as well. but I think we should work on a driver doc too.

**Comment by PowerShellCrack** [comment]
> Well it seems to find the drivers, but now all computers I try imaging just sit at the Getting Ready screen after the PSDWizard runs once. Not sure why that is happening. Ugh. I mean I am trying to install Windows 11 Enterprise 24H2, but I cannot imagine that's the issue, I will go to 23H2 if I have any better luck on this part.

does 23H2 work with same drivers?

**Comment by rmiller-ch** [comment]
Well I went on vacation for a few days after I posted this.  I will have to test 23H2 when I get back to the office on Tuesday, November 12.  I am pretty sure it worked with 24H2, but my problem now lies with domain join credentials.  I will know more this week as I ran out of time last week.

**Comment by rmiller-ch** [comment]
Just to circle back around.  I got everything working with the drivers and Windows 11 (24H2).  System is solid, @PowerShellCrack your directions filled in some of my gaps.  I appreciate all the help I received.

**Comment by PowerShellCrack** [comment]
@rmiller-ch are you ok with closing this 

@GeoSimos
